game-core/src/test/java/games/strategy/engine/lobby/client/login/LobbyPropertyFileParserTest.java

```diff
@@ -55,25 +52,20 @@ void parseWithSimpleCase() throws Exception {
     testProps.message = TestData.message;
     testProps.version = TestData.clientCurrentVersion;
 
-    final File testFile = newTempFile(testProps);
+    final String yamlContents = newYaml(testProps);
 
     final LobbyServerProperties result =
-        LobbyPropertyFileParser.parse(testFile, new Version(TestData.clientCurrentVersion));
+        LobbyPropertyFileParser.parse(yamlContents, new Version(TestData.clientCurrentVersion));
     assertThat(result.getHost(), is(TestData.host));
     assertThat(result.getPort(), is(Integer.valueOf(TestData.port)));
     assertThat(result.getServerMessage(), is(TestData.message));
     assertThat(result.getServerErrorMessage(), is(TestData.errorMessage));
   }
 
-  private static File newTempFile(final TestProps... testProps) throws Exception {
-    final File f = File.createTempFile("testing", ".tmp");
-    try (Writer writer = Files.newBufferedWriter(f.toPath(), StandardCharsets.UTF_8)) {
-      for (final TestProps testProp : Arrays.asList(testProps)) {
-        writer.write(testProp.toYaml());
-      }
-    }
-    f.deleteOnExit();
-    return f;
+  private static String newYaml(final TestProps... testProps) throws Exception {
```

Comment:
This method no longer throws a checked exception, so the exception specification can be removed.